compiler/infra/SimpleRegex.cpp

```diff
@@ -486,6 +487,57 @@ bool SimpleRegex::match(
    return result;
    }
 
+bool SimpleRegex::match(
+      TR::SimpleRegex *regex,
+      TR_ByteCodeInfo& bcinfo,
+      bool isCaseSensitive)
+   {
+   TR::Compilation *comp = TR::comp();
+   TR::StackMemoryRegion stackMemoryRegion(*comp->trMemory());
+   TR::StringBuf buf(stackMemoryRegion);
+
+   // Append the signature of the outermost method
+   buf.appendf("%s", comp->signature());
```

Comment:
I imagined that `#` would be prepended before every signature, including this one, to mark the beginning of the signature, and that we would pass a glob starting with `*#`, e.g. `{*#java/util/HashMap.get...}`. But as you have it here we can still do the same thing just by dropping that initial `#`: `{*java/util/HashMap.get...}`. It's highly unlikely that the specified signature will be a suffix of the signature of any other method. And it's rare, but `#` can occur in signatures, so it's not as though it would be guaranteed to be an inlined frame separator anyway.

Reply:
Hmmm.  Yes, I see now that including the initial `#` could help avoid spurious matches in situations where leading parts of one package name overlap those of another.  For instance, if someone made a modified copy of some open source project's code, they might have both an `org/eclipse/xyzzy/Foo.bar()V` and `com/example/org/eclipse/xyzzy/Foo.bar()V`.  Including the leading `#` would help avoid unexpected matches, so I will make that change.

As for the fact that `#` can occur in signatures, I understand the same is true of `@` if I'm not missing something.  I'm guessing you're OK with that, given their rarity and the fact that this is primarily intended for debugging or investigating performance issues.  If not, would you have any suggestions for alternative separators?  Perhaps sequences of characters that could not occur, such as `//` or `;;`?  Even then, such separators might work within the context of a Java virtual machine, but might still be problematic in the context of other programming language runtime environments.

Reply:
Oh, that's a good example! In that case, yes, thanks for including the leading `#`.

There's very little that we can safely assume will never occur in a signature, even if we're only talking about Java. In the broader context of OMR, that might be nothing instead of very little. So I think we have to settle for markers that are merely unlikely to appear, but I also think that should be fine for this use case

Reply:
Added a `#` before the outermost signature in commit ecc34b7791f6b8f6cfbb12a9a6f4e7a90047fa89